### PR TITLE
gh-114058: Fix flaky globals to constant test

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
The optimizer does not optimize global loads if the builtins dictionary has been modified more than 3 times. In this case, I suspect some other test is modifying the builtins dictionary before test_capi.test_opt is run. So the test for the optimizer fails if tests are run sequentially.

The solution is to run the test in the subprocess so it's not subject to shenanigans that other tests do :).

* Issue: gh-114058

### CODE REVIEW

**Comment by Fidget-Spinner** [comment]
Easy reproducer: thanks to @Eclips4 

```
./python -m test test___all__ test__locale test__opcode test__osx_support test__xxinterpchannels test__xxsubinterpreters  test_abc test_abstract_numbers  test_argparse  test_array test_asdl_parser test_ast test_asyncgen test_atexit test_audit test_augassign  test_augassign test_baseexception  test_bdb  test_bigaddrspace  test_bigmem  test_binascii  test_binop  test_bisect  test_bool  test_buffer test_bufio  test_builtin  test_bytes  test_bz2  test_c_locale_coercion  test_calendar  test_call  test_capi
```

**Comment by Eclips4** [comment]
> Easy reproducer: thanks to @Eclips4
> 
> ```
> ./python -m test test___all__ test__locale test__opcode test__osx_support test__xxinterpchannels test__xxsubinterpreters  test_abc test_abstract_numbers  test_argparse  test_array test_asdl_parser test_ast test_asyncgen test_atexit test_audit test_augassign  test_augassign test_baseexception  test_bdb  test_bigaddrspace  test_bigmem  test_binascii  test_binop  test_bisect  test_bool  test_buffer test_bufio  test_builtin  test_bytes  test_bz2  test_c_locale_coercion  test_calendar  test_call  test_capi
> ```

I found the more easier reproducer:
```
./python -m test test_argparse test_ast test_asyncgen test_capi
```

**Comment by Eclips4** [comment]
macOS failure is unrelated, see https://github.com/python/cpython/issues/115383 for more information

File: Lib/test/test_capi/test_opt.py [hunk 1]
```diff
@@ -749,20 +751,39 @@ def dummy(x):
...
         ex = get_first_executor(testfunc)
-        self.assertIsNotNone(ex)
+        assert ex is not None
         uops = {opname for opname, _, _ in ex}
-        self.assertNotIn("_LOAD_GLOBAL_BUILTIN", uops)
-        self.assertIn("_LOAD_CONST_INLINE_BORROW_WITH_NULL", uops)
+        assert "_LOAD_GLOBAL_BUILTINS" not in uops
+        assert "_LOAD_CONST_INLINE_BORROW_WITH_NULL" in uops
+        """))
+        self.assertEqual(result[0].rc, 0)
```
Comment (vstinner) [inline Lib/test/test_capi/test_opt.py]: Display stdout and stderr on error:

```suggestion
        self.assertEqual(result[0].rc, 0, result)
```

**Pull request merged**